src/interpreter/index.ts

```diff
@@ -122,7 +122,9 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
                 );
                 let statements = await componentScopeResolver.resolve(component);
                 interpreter.inSubEnv(subInterpreter => {
-                    subInterpreter.environment.setM(interpreter.environment.getM());
+                    let componentMPointer = new RoAssociativeArray([]);
+                    subInterpreter.environment.setM(componentMPointer);
```

Comment:
I don't really understand this change, what was wrong with setting `interpreter.environment.getM()` here?

Reply:
That was passing down the existing `mPointer` so then `top` was being added to the global mPointer and the script that created a component could access `m.top.*` of that component.

Reply:
For example:

```brightscript
sub main()
    ' create an object that uses m.top
    myComponent = createObject("roSGNode", "MTopWidget")

    ' m.top in this script should not refer to anything
    print m.top
end sub
```
Using `interpreter.environment.getM()` would mean that `m.top` in this script would print out the top value of the `MTopWidget` we created